python/scripts/load_data.py

```diff
@@ -20,208 +20,403 @@
+        Returns an interface object for the sql database
+        
+        :param manifest_row: a given row in the manifest
+        :param temp_filepath: the file path where PSV will be saved
+        """
+        # check for database manifest - create it if it doesn't exist
+        sql_manifest_exists = \
+            ingestionfunctions.check_or_create_sql_manifest(engine=self.engine)
+        logging.info("sql_manifest_exists: {}".format(sql_manifest_exists))
+
+        # configure database interface object and get matching manifest row
+        interface = HISql(meta=self.meta, manifest_row=manifest_row,
+                          engine=self.engine, filename=temp_filepath)
+        return interface
+
+    def _create_clean_psv(self, table_name, manifest_row, csv_reader,
+                          temp_filepath):
+        """
+        Cleans the data for the table name in the given manifest row, writes 
+        the clean data to PSV file, and then passes on that information so 
+        the database can be updated accordingly.
+        """
+        # get database interface and it's equivalent manifest row
+        sql_interface = self._configure_db_interface(
+            manifest_row=manifest_row, temp_filepath=temp_filepath)
+
+        sql_manifest_row = sql_interface.get_sql_manifest_row()
+
+        cleaner = self._get_cleaner(table_name=table_name,
+                                    manifest_row=manifest_row)
+        csv_writer = CSVWriter(meta=self.meta,
+                               manifest_row=manifest_row,
+                               filename=temp_filepath)
+
+        # clean the file and save the output to a local pipe-delimited file
+        # if it doesn't have a 'loaded' status in the database manifest
+        if csv_reader.should_file_be_loaded(
+                sql_manifest_row=sql_manifest_row):
+            print("  Cleaning...")
+            meta_only_fields = self._get_meta_only_fields(
+                table_name=table_name, data_fields=csv_reader.keys)
+            for idx, data_row in enumerate(csv_reader):
+                data_row.update(meta_only_fields)  # insert other field dict
+                clean_data_row = cleaner.clean(data_row, idx)
+                if clean_data_row is not None:
+                    csv_writer.write(clean_data_row)
+
+            csv_writer.close()
+
+            # write the data to the database
+            unique_data_id = manifest_row['unique_data_id']
+            self._update_database(table_name=table_name,
+                                  sql_interface=sql_interface)
+
+            if not self._keep_temp_files:
+                csv_writer.remove_file()
+
+    def _update_database(self, table_name, sql_interface):
+        """
+        Load the clean PSV file into the database
+        """
+        print("  Loading...")
+
+        # append new unique_data_id data instead of dropping entire table
+        # TODO: remove unnecessary code - relace table not needed?
```

Comment:
Yeah, we can remove the replace_table stuff. Instead, we'll tell it whether or not to replace the unique_data_id (i.e. by deleteing the rows).